packages/phoenix-evals/src/phoenix/evals/models/openai.py

```diff
@@ -279,27 +280,43 @@ def _init_rate_limiter(self) -> None:
         )
 
     def _build_messages(
-        self, prompt: MultimodalPrompt, system_instruction: Optional[str] = None
+        self, prompt: MultimodalPrompt, data_fetcher: Optional[Callable[[str], Audio]] = None, system_instruction: Optional[str] = None
     ) -> List[Dict[str, str]]:
         messages = []
-        for parts in prompt.parts:
-            if parts.content_type == PromptPartContentType.TEXT:
-                messages.append({"role": "system", "content": parts.content})
+        for part in prompt.parts:
+            if part.content_type == PromptPartContentType.TEXT:
+                messages.append({"role": "system", "content": part.content})
+            elif part.content_type == PromptPartContentType.AUDIO:
+                audio_object = data_fetcher(part.content)
```

Comment:
I don't super like the fact that we've plumbed an arbitrary callable down to this level, maybe I'm overthinking it